AdminNotice.php

```diff
@@ -484,10 +506,28 @@ public function isDismissed() {
 			}
 
 			if ($this->dismissionScope === self::DISMISS_PER_SITE) {
-				return (boolean)(get_option($this->getDismissOptionName(), false));
+				$dismissal = get_option($this->getDismissOptionName());
 			} else {
-				return (boolean)(get_user_meta(get_current_user_id(), $this->getDismissOptionName(), true));
+				$dismissal = get_user_meta(get_current_user_id(), $this->getDismissOptionName(), true);
 			}
+
+			$dismissal = json_decode($dismissal, true);
```

Comment:
I believe that `json_decode` can trigger a warning if the input is invalid or not a string, so I recommend adding `@` here.
```suggestion
			$dismissal = @json_decode($dismissal, true);
```

Reply:
I've committed an alternative.